src/indexer/merger.rs

```diff
@@ -303,6 +322,82 @@ impl IndexMerger {
         Ok(())
     }
 
+    // used to merge `u128` single fast fields.
+    fn write_u128_multi_fast_field(
+        &self,
+        field: Field,
+        fast_field_serializer: &mut CompositeFastFieldSerializer,
+        doc_id_mapping: &SegmentDocIdMapping,
+    ) -> crate::Result<()> {
+        let segment_and_ff_readers = self
+            .readers
+            .iter()
+            .map(|segment_reader| {
+                let ff_reader: MultiValuedU128FastFieldReader<u128> =
+                    segment_reader.fast_fields().u128s(field).expect(
+                        "Failed to find index for multivalued field. This is a bug in tantivy, \
+                         please report.",
+                    );
+                (segment_reader, ff_reader)
+            })
+            .collect::<Vec<_>>();
+
+        Self::write_1_n_fast_field_idx_generic(
+            field,
+            fast_field_serializer,
+            doc_id_mapping,
+            &segment_and_ff_readers,
+        )?;
+
+        let fast_field_readers = segment_and_ff_readers
+            .into_iter()
+            .map(|(_, ff_reader)| ff_reader)
+            .collect::<Vec<_>>();
+
+        let iter = || {
+            doc_id_mapping.iter_old_doc_addrs().flat_map(|doc_addr| {
+                let fast_field_reader = &fast_field_readers[doc_addr.segment_ord as usize];
+                let mut out = vec![];
```

Comment:
Please remove the allocation in each iteration of the Iterator.

Reply:
It's not so easy, we tried that in the other case unsuccessfully. It would be easier with an iterator instead of get_vals.

Reply:
> It's not so easy, we tried that in the other case unsuccessfully. It would be easier with an iterator instead of get_vals.

I sent you a solution before. You need an iterator implementation.

Reply:
I would rather keep the code simple and slightly slower here.